playbooks/app_dev/kdl.adoc

```diff
@@ -0,0 +1,174 @@
+|https://kubernetes.io/docs/concepts/services-networking/service/#headless-services[Cluster IP, ClusterIP: None]|HS a.k.a. Headless Service
+|https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport[Node Port]|NP
+|https://kubernetes.io/docs/concepts/services-networking/service/#type-loadbalancer[LoadBalancer]|LB
+|https://docs.openshift.com/container-platform/3.5/dev_guide/integrating_external_services.html#using-fqdn-2[External Name] (OpenShift only)|EN
+|https://kubernetes.io/docs/concepts/services-networking/service/#external-ips[External IP] |EIP
+|===
+
+At the top of the service there are the exposed ports. Same convention applies here as for the compute ports.
+
+The service should be connected to a compute object. This will implicitly define the service selector, so there is no need to have it indicated in the picture.
+
+If a service is allows traffic from the outside of the cluster to internal pods (such as for Load Balancer or Node Port or External IP) it should be depicted on the edge of the cluster.
+
+image::./images/edge-service.png[]
+
+Same concept applies to services that regulate outbound traffic (such as External Name), although in this case they would probably appear at the bottom of the openshift cluster rectangle.
+
+==== Ingresses 
+
+Ingresses can be indicated with a parallelogram as in the following picture:
+
+image::./images/ingress-template.png[]
+
+An ingress shows the ingress name and optionally the host exposed. An ingress will be connected to a service (the same rules apply to OpenShift routes).  
+Ingresses are always shown at the edge of the openshift cluster. 
+
+image::./images/edge-ingress.png[]
+
+=== Storage
+
+Storage is used to indicate persistent volumes. The color of storage is blues and it’s shape is a bucket deployed as the following picture:
+
+image::./images/storage-template.png[]
+
+Storage should indicate the persistent volume name and the storage provider (example NFS, gluster etc...). 
+Persistent Storage is always depicted at the edge of the cluster because it is a configuration pointing to an externally available storage. 
+￼
+image::./images/edge-storage.png[]
+
+== Putting it all together
+
+In this section we will go over an example of how this notation can be used to describe the deployment of an application. 
+Our application is an bank service application that uses a mariadb database as its datastore. 
+Here is the deployment diagram:
+
+image::./images/mariadb-example.png[]
+
+
+Notice that the mariadb pod uses StatefulSet and a persistent volume for its data. This pod is not exposed externally to the cluster, but its service is consumed by the BankService app. +
+The BankService app is a stateless pod controlled by a deployment config which has a secret with the credentials to access the database. It also has a service and a route so that it can accept inbound connection from outside the cluster.
+
```

Comment:
In general, I really like having this type of Language to work with.  I've used it a couple times, naively, and it works.  At some point, I'd consider getting slightly more formal, and, maybe bouncing this against other Visual Languages with similar aims

[Creole](http://plantuml.com/creole)
[PlantUML](http://plantuml.com/)
[Salt](http://plantuml.com/salt)

Reply:
was there any change that you wanted me to do here?